iocore/dns/DNS.cc

```diff
@@ -424,12 +430,27 @@ DNSEntry::init(const char *x, int len, int qtype_arg, Continuation *acont, DNSPr
 }
 
 /**
+ Open UDP and/or TCP connections based on dns_conn_mode
+ */
+
+void
+DNSHandler::open_cons(sockaddr const *target, bool failed, int icon)
+{
+  if (dns_conn_mode != DNS_CONN_MODE::TCP_ONLY) {
+    open_con(target, failed, icon, false);
+  }
+  if (dns_conn_mode != DNS_CONN_MODE::UDP_ONLY) {
```

Comment:
The TCP connections we open are persistent. And the number is small since only one per name server.
One reason I open it in the beginning instead of right before we need it is that establishing TCP connection is relatively slow. So we should open it as soon as possible to decrease the latency of DNS querying.